eng/versioning.targets

```diff
@@ -89,6 +89,7 @@
   <ItemGroup Condition="'$(IsTestProject)' != 'true'">
     <SupportedPlatform Condition="'$(TargetPlatformIdentifier)' == 'illumos'" Include="illumos" />
     <SupportedPlatform Condition="'$(TargetPlatformIdentifier)' == 'solaris'" Include="Solaris" />
+    <SupportedPlatform Condition="'$(TargetPlatformIdentifier)' == 'haiku'" Include="Haiku" />
```

Comment:
This shouldn't be needed as we don't plan on enabling managed libraries targeting haiku. Basically, we don't want to enable the following: `net8.0-haiku`.

Reply:
Not in this pull request, but I intend to open a PR in the future with the implementation of managed libraries for Haiku: https://github.com/trungnt2910/dotnet-runtime/tree/dev/trungnt2910/haiku-lib

Does it actually break CI and/or any internal conventions?

Reply:
> Does it actually break CI and/or any internal conventions?

It is tied to [platform exclusion](https://github.com/dotnet/designs/blob/7d0be161bfb55117543f2833b645e089b646f8ab/accepted/2020/platform-exclusion/platform-exclusion.md) infrastructure. @ViktorHofer, CMIIR, but I think if Haiku library support will be added subsequently, it shouldn't break anything?

Reply:
For https://github.com/trungnt2910/dotnet-runtime/commit/59942abba4ec988a45016c3f18bf3d41fbd6f4d1, we don't need the code above. System.Private.CoreLib and doesn't use `TargetPlatformIdentifier`. Any other library under src/libraries would require haiku to be part of the runtime.json RID graph and have the above `SupportedPlatform` attribute.

But as mentioned, I doubt that we want managed components to target haiku. We should try to make them use the existing Unix/Linux implementations and if needed use runtime checks to differentiate for haiku. Adding a new platform to src/libraries isn't trivial and bloats the entire build graph for everyone.

Reply:
> This would just bloat up the IL footprint for everyone.

It wouldn't as we invoke the linker on our inbox assemblies (which are part of the Microsoft.NETCore.App shared framework) so that non OS applicable code gets removed. For assemblies that ship via nuget packages (i.e. `System.Formats.Cbor`) we don't want to introduce new build configurations as the cost of these is higher than the benefit (i.e. package size, build complexity, etc.).

Self-contained customer apps use the linker as well which results in unnecessary IL being removed.

> I thought that only a few platforms listed in src/installer/pkg/projects/netcoreappRIDs.props are officially built? As for local builds I don't think that all libraries are built, just the one for the OS passed in ./build.sh?

This only applies to shared framework and native packages. For out-of-band packages, anything under src/libraries is included in the produced packages. With every additional managed build platform we pay that cost in CI and locally when invoking the `-allconfigurations` build.

Reply:
cc @ericstj @jkotas

Reply:
> we don't need the code above. System.Private.CoreLib and doesn't use `TargetPlatformIdentifier`.

I intend to add implementations for other managed libraries in the near future, such as the `System.Diagnostic.Process` (currently having some WIP code for that on a local repo) required to get `dotnet/sdk` running on Haiku (otherwise it would just throw a `PlatformNotSupportedException` on starutup).

Reply:
> we don't want to introduce new build configurations as the cost of these is higher than the benefit (i.e. package size, build complexity, etc.).

I agree with you that it would be nice to refactor libraries to have one configuration, shared by all Unix flavors. It would be independent project from the Haiku port.

This PR is following what we do for all other Unix flavors today. I do not see a problem with it.